lib/pbench/test/functional/agent/cli/commands/tools/test_clear_tools.py

```diff
@@ -217,238 +273,109 @@ def test_clear_tools_test70(monkeypatch, agent_config, pbench_run, pbench_cfg):
-        else:
-            assert tool.exists() is True
-
-    command = ["pbench-clear-tools", "--name=pidstat,mpstat,iostat"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert b"" == out
-    for tool in tools_existence:
-        assert tool.exists() is False
-
-
-@pytest.mark.parametrize("groups", ["custom1", "custom1,custom2"])
-@pytest.mark.parametrize("remotes", ["remote1", "remote1,remote2"])
-def test_clear_tools_test75(
-    monkeypatch, agent_config, pbench_run, pbench_cfg, groups, remotes
-):
-    """ Attempt to clear all tools in non-default groups, by name, separated by
-        comma"""
-    groups_shouldnt_exists = []
-    for group in groups.split(","):
-        another_group = pbench_run / f"tools-v1-{group}"
-        another_group.mkdir(parents=True)
-        groups_shouldnt_exists.append(another_group)
-        for remote in remotes:
-            remote_host = another_group / f"{remote}.example.com"
-            remote_host.mkdir(parents=True, exist_ok=True)
-            pidstat7_tool = remote_host / "pidstat"
-            pidstat7_tool.touch()
-            mpstat7_tool = remote_host / "mpstat"
-            mpstat7_tool.touch()
-
-    command = ["pbench-clear-tools", f"--group={groups}", "--name=pidstat,mpstat"]
-    out, err, exitcode = pytest.helpers.capture(command)
-    assert b"" == out
-    for group in groups_shouldnt_exists:
-        assert group.exists() is False
-
-
-@pytest.mark.parametrize(
-    "groups", ["default", "default,custom1", "default,custom1,custom2"]
-)
 @pytest.mark.parametrize(
-    "remotes",
-    ["remote1", "remote1,remote2", "remote1,remote2,remote3", "remote2,remote4"],
+    "tools", ["pidstat,mpstat,iostat", "mpstat,vmstat"],
 )
-def test_clear_tools_test76(
-    monkeypatch, agent_config, pbench_run, pbench_cfg, groups, remotes
+def test_clear_tools_test86(
+    monkeypatch,
+    tmp_path,
+    agent_config,
+    pbench_run,
+    pbench_cfg,
+    tools_configuration,
+    groups,
+    remotes,
+    tools,
 ):
-    """ Attempt to clear groups, by name, registered on multiple
-    hosts"""
-    tools_existence = []
-    for group in groups.split(","):
-        another_group = pbench_run / f"tools-v1-{group}"
-        another_group.mkdir(parents=True)
-
-        for remote in remotes.split(","):
-            fubar7_1_host = another_group / f"{remote}.example.com"
-            fubar7_1_host.mkdir(parents=True, exist_ok=True)
-            pidstat7_1_tool = fubar7_1_host / "pidstat"
-            pidstat7_1_tool.touch()
-            tools_existence.append(pidstat7_1_tool)
-
+    """ Attempt to clear tools at various group-remote-tool locations where the
+    combination may not present on the hosts"""
     command = [
         "pbench-clear-tools",
-        f"--group={groups}",
-        "--remote=remote2.example.com,remote3.example.com",
+        f"--groups={groups}" if groups else "",
+        f"--remotes={remotes}" if remotes else "",
+        f"--name={tools}" if tools else "",
```

Comment:
Unlike test 85, you don't have any empty values in any of the three parametrizations, so the ternary is pointless. Or did you intend to test against empty values?

Reply:
Yeah I copied it from the test-85, I should remove this ternary operations. I am not intending to test this against empty values as we are already doing it in test 85. This test is just to see if we get the intended message when one of the group/remote/tool mentioned in the command is not present on the host.

Reply:
Rather than removing the ternaries, perhaps we should be adding empty values to the permutations?  (We just have to filter out the cases which would pass.)

Reply:
Well, if any of these options are empty, the command will try to remove all the directories/files at that level. for example if the `remotes` is empty then it will remove all the remote directories under the group mentioned in the command, it wont generate this message that we are trying to test in this testcase `No remote host <remote>`